File: okj/szoftverfejleszto/pilotak/c#/Pilota.cs
using System;

public class Pilota {
    public const int URES_RAJTSZAM = -1;

    public readonly string nev;
    public readonly DateTime szuletesiDatum;
    public readonly string nemzetiseg;
    public readonly int rajtszam;

    public Pilota(string line) {
        var split = line.Split(';');

        nev = split[0];
        szuletesiDatum = DateTime.Parse(split[1]);
        nemzetiseg = split[2];
        rajtszam = line[line.Length - 1] == ';' ? URES_RAJTSZAM : int.Parse(split[3]);
    }
}

File: okj/szoftverfejleszto/pilotak/java/Pilota.java
import java.time.*;
import java.time.format.*;

public class Pilota {
    public static final int URES_RAJTSZAM = -1;
    private static final DateTimeFormatter DATUM_FORMAT = DateTimeFormatter.ofPattern("yyyy.MM.dd");

    public final String nev;
    public final LocalDate szuletesiDatum;
    public final String nemzetiseg;
    public final int rajtszam;

    public Pilota(String line) {
        var split = line.split(";");

        nev = split[0];
        szuletesiDatum = LocalDate.parse(split[1], DATUM_FORMAT);
        nemzetiseg = split[2];
        rajtszam = line.charAt(line.length() - 1) == ';' ? URES_RAJTSZAM : Integer.parseInt(split[3]);
    }
}